File: unidbg-api/src/main/java/com/vortexdbg/thread/Waiter.java
package com.vortexdbg.thread;

import com.vortexdbg.Emulator;
import com.vortexdbg.signal.SignalTask;

public interface Waiter {

    boolean canDispatch();

    void onContinueRun(Emulator<?> emulator);

    void onSignal(SignalTask task);
}

File: vortexdbg-api/src/main/kotlin/com/vortexdbg/thread/Waiter.kt
package com.vortexdbg.thread

import com.vortexdbg.Emulator
import com.vortexdbg.signal.SignalTask

interface Waiter {

    fun canDispatch(): Boolean

    fun onContinueRun(emulator: Emulator<*>)

    fun onSignal(task: SignalTask)
}
